### ISSUE TITLE
Emulator crash when attempting to play Kingdom Hearts Final Mix

### ISSUE THREAD

**Issue by deadair3210** [PCSX2/pcsx2#873]
Version - PCSX2 1.3.1-20151004230014
Specs - i5 2500k @ 4.0GHz, GTX 660, 11 GB RAM
Description - While playing Kingdom Hearts Final Mix, the emulator has a chance of crashing randomly.
Procedure - Play Kingdom Hearts Final Mix(NTSC-J), emulator will crash at a random time
If its not a random crash, I dont know what the cause would be. 
Log - https://gist.github.com/deadair3210/9a0dd1d35e5fb6273db5
PCSX2_vm,ini - https://gist.github.com/deadair3210/dc306b3da75e802c653b
GSDX.ini - https://gist.github.com/deadair3210/9c5744f8b83e9ce1f057
![capture](https://cloud.githubusercontent.com/assets/1053504/10272567/6e9fee42-6ae7-11e5-9d57-9520bd8838c5.PNG)

**Comment by willkuer** [comment]
Can you reproduce it using the file memory card system?

**Comment by deadair3210** [comment]
Not sure, will try. I'm not sure its a memory card issue as the crash will happen no matter what is happening. Here is a link to a play through I tried to do a few minutes ago. Emulator crashed in the middle of the battle
https://youtu.be/qdLJgBGNZvw

**Comment by deadair3210** [comment]
Well, I'm not sure how, but converting to a memory card file actually has seemed to fix the problem. I guess the memory card folder isnt as stable as I thought it was. Thanks for the help.

**Comment by deadair3210** [comment]
Closing the issue, or should I leave it open and just change the name to something like Memory card folder causes random games to crash. Its been affecting other games as well, as the exact same thing has been happening to me in Persona 3 FES and converting to card file fixed it as well

**Comment by willkuer** [comment]
I would keep it open. Maybe @AdmiralCurtiss could have a look?

**Comment by willkuer** [comment]
I think there are some meaningful memory card messages in the log. At least meaningful for somebody familiar with the emulated memory card system. Kh seems to do quite often access the memory card.

Can you somehow give more specific instructions how to reproduce the issue?

**Comment by AdmiralCurtiss** [comment]
Folder Memory Card messages seem normal to me. The "memcardErase ??" message comes from here: https://github.com/PCSX2/pcsx2/blob/539a1767a3508baeb0348a78fc9a82155dcbd088/pcsx2/Sio.cpp#L335, I've seen that one a bunch but I've never seen it cause issues. It is quite possible we're not correctly emulating some Erase request, though, a good vector of investigation on that is probably MGS3 since that game does produce a lot of those messages with different data values in the log.

Curious why it only happens with the Folder card though.

**Comment by deadair3210** [comment]
I'm not sure why it would crash mid-battle, unless Final Mix accesses the memory card at times the original didn't

**Comment by deadair3210** [comment]
Come to think about it, I was about to level up when the game froze. Maybe Final Mix saves something whenever you level up. I'm not sure, as I can't read much Japanesese. Also, could the Folder code be having trouble with the name of the save since its a Japanese game?

**Comment by willkuer** [comment]
But you are sure it is reproducible? Folder memory card randomly crashes while file memory does work? There are no exceptions?

**Comment by deadair3210** [comment]
Going to play it more later to see if the problem is truly solved, but I managed to play a hour after I switched to card instead of folder with no crashes where before it would usually crash somewhere in the first 30 minutes. I have a pretty good feeling its something to do with the Folder card

**Comment by rz5** [comment]
@willkuer Isn't it reasonable to say, based on the screenshot, that the crash happened because of GSdx2-AVX.dll after dereferencing a null pointer?

**Comment by willkuer** [comment]
Actually I didn't thought that the folder memory card is involved. It is just default troubleshooting behaviour to revert all saves to default ones and checking if the problem persists. That's also the reason why I wanted him to confirm that the issue is really related to foldermemorycards. The only real hint was that strange erase message that was considered to be possibly unrelated.

**Comment by deadair3210** [comment]
Just played the game for the better part of two hours, switching to the card definitely fixed it somehow

**Comment by avih** [comment]
I think they're trying to understand whether or not this crash is related to the folder memory card. So it's not enough to know that the crash doesn't happen with normal memory cards, it's also required to know that it _consistently_ does crash with folder memory card.

So if you can find a situation where it crashes consistently with folder memory card, and then you switch to normal memory card and go through this situation few times and it never crashes, and then go back to folder memory card and it crashes again, and then back to normal cards and it doesn't crash, etc, then we will know the crash is most likely _because_ of the folder memory card.

Otherwise, maybe something else has changed which make it not crash anymore, and then maybe it's not the necessarily the fault of the folder memory card.

In other words, you must be sure that it always crashes with folder memory card, or at the very least that it only crashes with folder memory card, and that never crashes with normal memory card.

**Comment by AdmiralCurtiss** [comment]
Also, since you appear to have Visual Studio installed, it would help to play with a self-compiled version of PCSX2 from within Visual Studio, so that when it crashes, we get a nice stacktrace of where it crashed and what the memory looked like at the time.

**Comment by deadair3210** [comment]
Sorry, Im new to Visual Studio, how do i do that

**Comment by willkuer** [comment]
I never tried to attach a debugger to a running program. Why don't you compile and debug the program within visual studio? As you have the code this would be probably easier.

**Comment by deadair3210** [comment]
So just compile the program, and when the emulator crashes, just hit debug?

**Comment by willkuer** [comment]
No. Just compile and run the program using the debug option. (I think F5 or F6 is the shortcut) Visual studio should then run the program. At some point some exception will happen and visual studio will send you to the file and line that is the issue.

**Comment by deadair3210** [comment]
Ok
Will do

**Comment by deadair3210** [comment]
Still crashes when trying to start the emulation
I open the sln file and let it load
after it loads i hit f5 and it builds and automatically starts the emulator
after placing bios, memcard, and plugins in the correct folders and hitting boot full the emulation window comes up and then throws an exception

**Comment by willkuer** [comment]
Yes but now you should be able to get a more verbose exception. Mostly important is the stack trace. So next time you get the exception click on details or something like that (I am used to visual c# and am not exactly sure how it will be for visual c++ but I guess similar)

**Comment by AdmiralCurtiss** [comment]
I've had a self-compiled PCSX2 crash on boot in Windows, it seems random and recompiling from scratch (Clean & Build) usually fixes it. I think it has something to do with plugins not loading right? What does the PCSX2 window say before it crashes?

**Comment by deadair3210** [comment]
Tried a Clean and then Build, same thing happens. The last thing in the log before the emulator crashes is SysExecute

**Comment by gregory38** [comment]
Use dev version to avoid some gsdx assertions. Exceptions in the recompilers are normal (but PITA). Just ignore them.

**Comment by deadair3210** [comment]
ok, when i get a chance i will attempt to reproduce the error with the latest git version

**Comment by Vanastick** [comment]
I've been having the same problem with the game, even in patched versions. I saw that he solved the problem by playing usig " file memory card system"... i don't know how to do that... so please help?! i would apreciate a lot.

**Comment by MrCK1** [comment]
Please redirect your questions to [https://forums.pcsx2.net/](url). Thank You!

**Comment by Vanastick** [comment]
I've been having the same problem with the game, even in patched versions. I saw that he solved the problem by playing usig " file memory card system"... i don't know how to do that... so please help?! i would apreciate a lot. Oh yeah... for me normally crashes when a enemy atacks...

**Comment by Vanastick** [comment]
said "404 not founded"

**Comment by MrCK1** [comment]
Sorry, I fixed the URL.

**Comment by deadair3210** [comment]
Don't have the specific build anymore so can't really test. If I remember correctly it didn't happen anymore but I don't clearly remember. Sorry

**Comment by MrCK1** [comment]
OK, I'll close it for now and it can be reopened if somebody encounters the issue.
